### DISCUSSION TITLE
DiepIO Related Discussions

### DISCUSSION THREAD

**Discussion by abcxff** [ABCxFF/diep-leftovers#1] [General]
First discussion

**Comment by abcxff** [comment]
Test - I've never tried this before lol

**Reply by Excigma** [reply]
poggers

**Reply by endless7z** [reply]
gg

**Comment by Colonizor48** [comment]
Hi

**Reply by abcxff** [reply]
> There are a lot of things i wana mess around with.

I don't think he'd do this kind of stuff - and I think he's been busy with college (tho he hasn't said he's in college right now) so he has slow response times

**Comment by abcxff** [comment]
I can't really describe how much information we've gathered in the past few months lol, so if there is anything you need just lmk

**Comment by Colonizor48** [comment]
Alright. Lets come back to that latter i guess. But lets focus on diep.

**Comment by abcxff** [comment]
![image](https://user-images.githubusercontent.com/79597906/109374838-3d0ac980-7886-11eb-9148-fe47a7f76af7.png)
Here's a sample of the memory modifications we're able to do right now (we have full control over entities in the memory, and in the data going through the websocket)

**Comment by abcxff** [comment]
Only thing still in question is how the game renders stuff, but besides that basically everything else has become clear - and since the game rebuilds 6:49 GMT every Sunday, we have developed systems to update our memory, packet, and wasm related code for us

**Comment by Colonizor48** [comment]
Allright that is some really good progress. Keep im mind i dont know much about code. But i want to learn. I have a bit of c++ experience and know the basics of programing

**Comment by abcxff** [comment]
Ah cool, you might want to check out some more javascript related tutorials, since thats the best (and only) way to interact with client stuff at the moment, and familiarize yourself with WebAssembly aka wasm - its how the game runs and knowing how specific instructions work is pretty useful

**Comment by Colonizor48** [comment]
Allright. I still have flashbacks to trying to learn x86 assembly as my first programing language. I thought that because the lines of code were short individually it would be easy. Boy was i wrong.

**Comment by abcxff** [comment]
lol, yeah x86 stuff is complex and simple at the same time, I don't think I'll ever spend much time on it though

**Comment by Colonizor48** [comment]
But now that i understand code a bit better i might be able to learn webassembly. I'll try my best. And brush up on my c++ aswell. I dont know how to do anything with graphics but when were making a server graphics probably dont matter much. But i dont know how to do internet stuff in c++ either. But from what i've seen it seems easy enough.

**Comment by abcxff** [comment]
Well good luck on your stuff, its been nice meeting you - getting late for me so I'm gonna get off. If you ever a chance to get on discord, I (and others) are on cx88's server as well as spade-squad official. Cya

**Comment by Colonizor48** [comment]
Allright cya and thank you. But one more thing

**Comment by Colonizor48** [comment]
Are we going to try to download the server software from a diep.io server or try to make our own server?

**Comment by abcxff** [comment]
We are just working on reversing the client, I have made some progress on a custom diep server - though it has no physics nor physical entities. And retrieving server source would not be legal - and the dev himself has said this, he is (inactive) in the florr.io discord.

**Comment by Colonizor48** [comment]
Allright. Thank you and have agood day.

**Comment by Colonizor48** [comment]
For when you next see this. Is the picture you showed me of the polygon on the sandbox server just a visual thing or does it influence the server? Also is it possible to have a player controled boss like in the pictures of the old repo?

**Reply by diepiodiscord** [reply]
It's client-sided/visual only, like inspect element

That player controlled boss was a screenshot by Zeach

**Reply by Colonizor48** [reply]
> 
> 
> It's client-sided/visual only, like inspect element
> 
> That player controlled boss was a screenshot by Zeach

thanks for the info. But my question is could there be a way to send an upgrade packet to upgrade to a tank that normally could not be upgraded into?

**Reply by abcxff** [reply]
> thanks for the info. But my question is could there be a way to send an upgrade packet to upgrade to a tank that normally could not be upgraded into?

There is not F, the game is very well made and there are little to no loop holes such as these

**Comment by Colonizor48** [comment]
Also tell CX that if they don't want to release the source code due to leaking beta keys then to try implimenting a hash system for beta keys. If if he doesnt want to then do it anyway for security reasons.

**Comment by Colonizor48** [comment]
Though honestly i dont see the point of having dedicated beta testers. Just make an "experimental" server that anyone can join and play upcomming tanks/gamemodes (there would be a seperate server for  playing as bosses and stuff) To get as much community feedback as possible. And you dont have to do a thing other then 5 minutes code changes to make the server and repuropusing some random server no one plays on into the experimental server.

**Comment by abcxff** [comment]
If you join the arras disc, you'll understand the reasoning behind all these things

**Comment by supahero1** [comment]
CX ain't publishing arras.io's code anytime soon, for his own reasons. He doesn't look for developers too so don't think about secretly stealing it or something. Diep.io's code will not be downloaded from anywhere or published by Zeach, and reversing the wasm files to C does pretty much nothing since it's the code of the client, not the server. If you want to write your own server then you need to reverse engineer the protocol 1:1. We have pretty much done it over the years, ABC thought about creating a custom server, idk how about the progress.

**Comment by Colonizor48** [comment]
Then CX should probably know he already leaked the source code for a relitivly recent version of arras.io. https://glitch.com/edit/#!/mazearrasthingy?path=OPENSHIFT.md%3A1%3A0. Yes i have coppied this for personal use. No i do not intend to make it public after this is deleted unless CX or another dev tells me its okay. Arras should already be open source due to the GPLv3 licence nephh just giving permission doesn't nullify the GPLlv3 license. But there would be a way to transition to closed source via nephhh just updating the repo and replacing it with Some other license and nulifying the previous one. But then again i'm no lawyer nor expert on the GPLv3.. But no one wants to be that guy at the party. But just updating the server template to the latest version of arras would be quite nice. I dont mean to start drama. But CX should probably delete the glitch repo if he wants the source to be private.

**Comment by Colonizor48** [comment]
If i were him i would try to transition to opensource or sort out the license. Because just because i wont sue him doesnt mean someone else won't. Some guy with personal beef or some person wanting money might try to sue. Or nephhh might happen to change his mind.

**Comment by abcxff** [comment]
Join us on discord, Im just going to keep this discussion for diep related stuff if possible. But just to wrap up, CX has full rights to his server code - you have no reason to sue him.
> Commercial use
 Modification
 Distribution
 Patent use
 Private use
 
The use of arras code for CX's arras.io do not violate these terms. Restating, there is no reason to sue, no valid argument to sue, and its just not worth it - especially since CX is a chill guy who has his own rights to his own code. Now lets just keep it on-topic though for future comments (you can write a reply or something if you disagree but I don't care to talk about this anyway lol)

**Reply by Colonizor48** [reply]
Yes lets limit that to a diffrent discussion. And for the record i'm not that kind of person. I'm just worried someone else might. If arras.io grew bigger then someone might try to pull it off. And this is exclusivly about the whole public source code thing. Or nephhh might have second thoughts. Anyway back to diep.

**Reply by Colonizor48** [reply]
Anyway we know all the tank ids right? Could there be a way to send a "fake" upgrade packet to the server to upgrade to an arena closer or somthing?

**Reply by abcxff** [reply]
https://github.com/ABCxFF/diep-leftovers/discussions/1#discussioncomment-414108 oops, just responded here

**Comment by Colonizor48** [comment]
Just out of curiosity what happens if you try?

**Reply by abcxff** [reply]
Nothing, just ignores it

**Comment by Colonizor48** [comment]
Allright. Another question. How do packets for sandbox commands work?

**Reply by abcxff** [reply]
Ill release a packet shuffler sometime later in this repo I think - all packets are shuffled / unshuffled atm so you need to hook into the wasm to retrieve anything worth while

**Reply by abcxff** [reply]
Or unshuffle using your own unshuffler

**Reply by abcxff** [reply]
Unless you mean the game console thing, in that case-  everything done there (with the exception of that spawn command) is all clientside stuff

**Reply by diepiodiscord** [reply]
They are bitflags in the outgoing 01 packet, CX already explained it on his repo

**Comment by abcxff** [comment]
Ill be dropping a packet hook for this update in `scripts/` Usage will be commented in

**Comment by Colonizor48** [comment]
allright sorry i've been quite busy with school lately. and the rest of the time i've just needed for leisure.

**Comment by endless7z** [comment]
hi abc
